src/lavinmq/clustering/client.cr

```diff
@@ -238,6 +238,7 @@ module LavinMQ
         @http_proxy.try &.close
         @files.each_value &.close
         @data_dir_lock.release
+        @socket.close
```

Comment:
socket needs to be a instance variable so that we can stop it from anohter fiber